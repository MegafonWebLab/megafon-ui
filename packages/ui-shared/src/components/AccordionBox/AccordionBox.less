@import '~@megafon/ui-core/styles/base.less';
@accordion: e('.mfui-beta-accordion');
@property: e('.mfui-beta-property');
@b: e('.mfui-beta-accordion-box');

@{b} {
    & + & {
        @{accordion} {
            border-top: none;
        }
    }

    @{property}:first-child {
        border-top: none;
    }
}
