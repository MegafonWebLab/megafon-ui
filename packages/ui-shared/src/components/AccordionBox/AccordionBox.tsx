import * as React from 'react';
import * as PropTypes from 'prop-types';
import { cnCreate, Grid, GridColumn, Accordion } from '@megafon/ui-core';
import './AccordionBox.less';

export interface IAccordionBox {
    /** Заголовок аккордеона */
    title: string;
    /** Состояние аккордеона, заданное извне */
    isOpened?: boolean;
    /** Центрирование по горизонтали для расширения 1280+ */
    hCenterAlignWide?: boolean;
    /** Вертикальные отступы */
    hasVerticalPaddings?: boolean;
    /** Дополнительный класс для корнеовго элемента */
    className?: string;
    /** Дополнительные классы для корневого и внутренних элементов */
    classes?: {
        root?: string;
        collapse?: string;
    };
    /** Обработчик клика */
    onClickAccordion?: (isOpened: boolean, title: string) => void;
}

const cn = cnCreate('mfui-beta-accordion-box');
const AccordionBox: React.FC<IAccordionBox> = ({hCenterAlignWide = false, ...props }) => (
    <div className={cn()}>
        <Grid hAlign={hCenterAlignWide ? 'center' : 'left'}>
            <GridColumn wide="8">
                <Accordion {...props} />
            </GridColumn>
        </Grid>
    </div>
);

AccordionBox.propTypes = {
    title: PropTypes.string.isRequired,
    isOpened: PropTypes.bool,
    hCenterAlignWide: PropTypes.bool,
    hasVerticalPaddings: PropTypes.bool,
    className: PropTypes.string,
    classes: PropTypes.shape({
        root: PropTypes.string,
        collapse: PropTypes.string,
    }),
    onClickAccordion: PropTypes.func,
};

export default AccordionBox;
