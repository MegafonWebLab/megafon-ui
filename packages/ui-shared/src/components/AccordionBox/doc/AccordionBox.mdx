---
name: AccordionBox
menu: Конструктор
route: components/accordionBox
---

import { Props } from 'docz';
import AccordionBox from '../AccordionBox';
import Example from './AccordionBox.example.mdx';

# AccordionBox

<CopyToClipboardBox text="import { AccordionBox } from '@megafon/ui-shared';" />

<DoczTabs>
    <Example />
    <Props of={AccordionBox} />
</DoczTabs>
