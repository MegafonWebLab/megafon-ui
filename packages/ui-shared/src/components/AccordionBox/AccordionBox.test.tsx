import * as React from 'react';
import { shallow } from 'enzyme';
import AccordionBox, { IAccordionBox } from './AccordionBox';

const props: IAccordionBox = {
    title: 'Test',
    isOpened: false,
    hasVerticalPaddings: true,
    className: 'className',
    classes: {
        root: 'rootClass',
        collapse: 'collapseClass',
    },
    onClickAccordion: jest.fn(),
};

describe('<AccordionBox />', () => {
    it('render AccordionBox', () => {
        const wrapper = shallow(
            <AccordionBox {...props}>
                <div>test</div>
            </AccordionBox>
        );

        expect(wrapper).toMatchSnapshot();
    });

    it('render AccordionBox with hCenterAlignWide=true', () => {
        const wrapper = shallow(
            <AccordionBox {...props} hCenterAlignWide>
                <div>test</div>
            </AccordionBox>
            );

        expect(wrapper).toMatchSnapshot();
    });
});
