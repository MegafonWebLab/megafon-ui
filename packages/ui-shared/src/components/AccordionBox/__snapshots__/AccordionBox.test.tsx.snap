// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<AccordionBox /> render AccordionBox 1`] = `
<div
  className="mfui-beta-accordion-box"
>
  <Grid
    hAlign="left"
  >
    <GridColumn
      wide="8"
    >
      <Accordion
        className="className"
        classes={
          Object {
            "collapse": "collapseClass",
            "root": "rootClass",
          }
        }
        hasVerticalPaddings={true}
        isOpened={false}
        onClickAccordion={[MockFunction]}
        title="Test"
      >
        <div>
          test
        </div>
      </Accordion>
    </GridColumn>
  </Grid>
</div>
`;

exports[`<AccordionBox /> render AccordionBox with hCenterAlignWide=true 1`] = `
<div
  className="mfui-beta-accordion-box"
>
  <Grid
    hAlign="center"
  >
    <GridColumn
      wide="8"
    >
      <Accordion
        className="className"
        classes={
          Object {
            "collapse": "collapseClass",
            "root": "rootClass",
          }
        }
        hasVerticalPaddings={true}
        isOpened={false}
        onClickAccordion={[MockFunction]}
        title="Test"
      >
        <div>
          test
        </div>
      </Accordion>
    </GridColumn>
  </Grid>
</div>
`;
